modules/contentbox/modules/contentbox-admin/modules/contentbox-filebrowser/handlers/Home.cfc

```diff
@@ -280,44 +284,59 @@ component{
 			event.renderData( data=data, type="json" );
 			return;
 		}
-
-		// clean incoming path and names
 		rc.path = cleanIncomingPath( URLDecode( trim( rc.path ) ) );
-		if( !len( rc.path ) ){
-			data.errors = true;
-			data.messages = $r( "messages.invalid_path@fb" );
-			event.renderData( data=data, type="json" );
-			return;
-		}
-
-		// Traversal Security
-		if( NOT isTraversalSecure(prc, rc.path) ){
-			data.errors = true;
-			data.messages = $r( "messages.traversal_security@fb" );
-			event.renderData( data=data, type="json" );
-			return;
+		
+			if( !len( rc.path ) ){
+				data.errors = true;
+				data.messages = $r( "messages.invalid_path@fb" );
+				event.renderData( data=data, type="json" );
+				return;
+			}
+		rc.pathsArray = ListToArray(rc.path);
+		if(fileExists(expandPath('\modules\contentbox\content\download.zip')))
+			fileDelete(expandPath('\modules\contentbox\content\download.zip'));
+		if(arrayLen(rc.pathsArray) > 1){
+			for(i=1;i<=arrayLen(rc.pathsArray);i++){
+				// Traversal Security
+				if( NOT isTraversalSecure(prc, rc.pathsArray[i]) ){
+					data.errors = true;
+					data.messages = $r( "messages.traversal_security@fb" );
+					event.renderData( data=data, type="json" );
+					return;
+				}
+				cfzip(action="zip",source="#rc.pathsArray[i]#",file="#expandPath('\modules\contentbox\content\download.zip')#");
```

Comment:
Maybe it would be best to use `cfzipParams` instead so you can pass all the files in one shot
`
```
cfzip( action="zip", file = "" ){
  for( var thisFile in rc.pathsArray ){
    cfzipParam()
 }
}
```